### PR TITLE
Fix some clippy warnings

### PR DESCRIPTION
**Pull request by remilauzier**

### CODE REVIEW

**Review by dhardy — COMMENTED** [review]
Thanks for the effort. This looks good, maybe without the mentioned `or_else`.

I usually don't bother with Clippy, because there are too many false-positives / disagreeable suggestions. I had a quick skim, and there are several more sensible suggestions, some stuff it splits hairs over (I don't get what's wrong with `v.len() == 0`), and some stuff that's outright wrong:
```
error: this loop never actually loops
   --> src/widget/view/matrix_view.rs:513:28
    |
513 |               let response = 'outer: loop {
    |  ____________________________^
514 | |                 // We forward events to all children, even if not visible
515 | |                 // (e.g. these may be subscribed to an UpdateHandle).
516 | |                 for child in self.widgets.iter_mut() {
...   |
523 | |                 return Response::Unhandled;
524 | |             };
    | |_____________^
```
... an ugly pattern, but show me a better one (there's been *lots* of discussion on this, e.g. [here](https://internals.rust-lang.org/t/pre-rfc-break-with-value-in-for-while-loops/11208)).

File: src/adapter/label.rs [hunk 1]
```diff
@@ -96,7 +96,7 @@ impl<W: Widget, D: Directional> Layout for WithLabel<W, D> {
         if !self.rect().contains(coord) {
             return None;
         }
-        self.inner.find_id(coord).or(Some(self.id()))
+        self.inner.find_id(coord).or_else(|| Some(self.id()))
```
[the remark this answers is not in the document]
Comment (remilauzier) [reply src/adapter/label.rs]: I put a clippy allow for that.

**Comment by remilauzier** [comment]
I correct what you want. Don't hesitate if you want to silence others warning or change something else.

File: src/adapter/label.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (remilauzier) [reply src/adapter/label.rs]: For ``v.len() == 0`` clippy said it's much faster to check for emptiness with ``.is_empty``.
https://rust-lang.github.io/rust-clippy/master/index.html#len_zero

[the remark this answers is not in the document]
Comment (dhardy) [reply src/adapter/label.rs]: > Some structures can ...

It depends on the data type, e.g. a linked list or map structures *may* be slow here. `Vec` is always fast.

**Review by dhardy — APPROVED** [review]
Looks good, thanks. Shall I merge or are you intending to add to this?

**Comment by remilauzier** [comment]
Yeah, you can merge.

**Pull request merged**